crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -154,20 +154,25 @@ impl SlidingSync {
         let mut sticky = self.inner.sticky.write().unwrap();
         let room_subscriptions = &mut sticky.data_mut().room_subscriptions;
 
+        let mut skip_sync_loop = false;
+
         for room_id in room_ids {
-            if let Some(room) = self.inner.client.get_room(room_id) {
-                room.mark_members_missing();
+            if let Entry::Vacant(entry) = room_subscriptions.entry((*room_id).to_owned()) {
```

Comment:
I would probably add a comment explaining what is going on here, from the PR description it is clear, but that might be hard to find again as time goes on.

Reply:
Indeed. I've improved to public doc, I've added an inline doc, and I've rebased the entire change.